### ISSUE TITLE
Unreliable httpd link causing CI failures

### ISSUE THREAD

**Issue by tobil4sk** [HaxeFoundation/neko#265]
On Mac, downloads of https://archive.apache.org/dist/httpd/httpd-2.4.29.tar.gz keep timing out.

There are [mirror sites](https://www.apache.org/dyn/closer.cgi/httpd/), however those only keep the latest releases.

**Comment by tobil4sk** [comment]
Relevant: https://github.com/apache/fluo-muchos/issues/273

**Comment by tobil4sk** [comment]
@andyli Any idea how we could set something like this up? The issue is that the regular download page only ever has the most recent release, and the archive page is rate limited and isn't intended for downloading via scripts.

Perhaps we can store a copy on github like we do with [ffmpeg for hashlink](https://github.com/HaxeFoundation/hashlink/commit/088b38323061d312c0351cfe5680743e0dcbba86).

**Comment by andyli** [comment]
Yeah, any mirror is fine as long as the MD5 checksum is the same.

We probably want to use [multiple `URL ...` in `ExternalProject_Add`](https://cmake.org/cmake/help/latest/module/ExternalProject.html?highlight=externalproject_add#command:externalproject_add), which are used by cmake one by one until one succeeded. That feature was introduced in CMake 3.7.0, so we need to bump the `cmake_minimum_required` at the top of `CMakeLists.txt` if we do that.

[According to repology](https://repology.org/project/cmake/versions), the current oldest supported Ubuntu LTS (18.04) provides CMake 3.10.2. Other distros have even newer CMakes, so should be fine.

**Comment by tobil4sk** [comment]
Alright, I made a separate PR to bump CMake: #267

Once that is merged I will update #266 to add the multiple URLs.

**Comment by tobil4sk** [comment]
I've updated changes #266 with the cmake changes. I just added the file uploaded here as a backup option in case the archive server blocks the request. So for local builds it will still use the archive. The md5 hash is the same as well.
